### PR TITLE
Rails 6.1 ruby 3.1

### LINKED ISSUE #389: Known incompatible with ruby 3.1 -- YAML load of symbols
While b-e may not be tested with ruby 3.0 (not sure), in fact I'm using it with ruby 3.0 just fine (doesn't mean all codepaths are clean necessarily). 

But trying to upgrade my app to ruby 3.1, I found out that b-e is definitely not compatible with ruby 3.1. 

Got an `ActionView::Template::Error (wrong number of arguments (given 2, expected 1))`, on:

```
ruby-3.1.2/lib/ruby/3.1.0/psych.rb:323:in `safe_load'
browse-everything (1.1.2) lib/browse_everything.rb:58:in `configure'
browse-everything (1.1.2) lib/browse_everything.rb:78:in `config'
browse-everything (1.1.2) lib/browse_everything/browser.rb:13:in `initialize'
browse-everything (1.1.2) app/services/browser_factory.rb:6:in `new'
browse-everything (1.1.2) app/services/browser_factory.rb:6:in `build'
browse-everything (1.1.2) app/controllers/browse_everything_controller.rb:123:in `browser'
browse-everything (1.1.2) app/controllers/browse_everything_controller.rb:128:in `browser'
```

https://github.com/samvera/browse-everything/blob/69a22a5ea17ef0f159987a904e5935bd50e59598/lib/browse_everything.rb#L58

I'm not sure what the second arg to safe_load in `YAML.safe_load(config_file_template.result, [Symbol])` is trying to do? But looks like a second arg is no longer supported?

Here's the commit that introduced it, with commit message "Allow browse_everything_providers.yml to contain symbols"

https://github.com/samvera/browse-everything/commit/11f01548a3092b9a43a0fa8f960f470ea565838d

It looks like the second arg is labelled `legacy_permitted_classes` in the docs: https://ruby-doc.org/stdlib-2.6.1/libdoc/psych/rdoc/Psych.html#method-c-safe_load

It may be that in ruby 3.1, you can no longer do a `YAML.safe_load` that returns deserialized data with symbols?  If browse_everything is depending on that behavior, that could be a somewhat more painful thing to fix.

### PR DESCRIPTION
**Pull request by jrochkind**
* Requires some Gemfile.extra business to get engine_cart to succesfully create a rails 6.1/ruby 3.1 project prior to `mail` gem 2.8.0 being released. https://bibwild.wordpress.com/2022/06/13/using-engine_cart-with-rails-6-1-and-ruby-3-1/

* Then we run into an actual 3.1 failure, with removal of deprecated method args for `YAML.safe_load`, need to use new non-deprecated form of arg.  New non-deprecated form is available as of ruby 2.6 -- and we already dropped ruby 2.5 from support/CI build matrix in #395 , so we're good without a conditional! 
   * Closes #389

### CODE REVIEW

**Comment by jrochkind** [comment]
Darnit, workaround for ruby 3.1/rails 6.1 with mail 2.8.0.rc1 seems to conflict with some of the engine)_cart business around step "Update sprockets to fix version incompatibilities between internal test app and initial bundle install"

It results in:

```
You have requested:
  mail >= 2.8.0.rc1

The bundle currently has mail locked at 2.7.1.
Try running `bundle update mail
```

Not sure why this was not a problem for me locally. It's possible that step is no longer necessary.... 

It's possible we should consider removing engine_cart entirely

**Comment by jrochkind** [comment]
Okay, there was a circleci step:

```
      - run:
         name: Update sprockets to fix version incompatibilities between internal test app and initial bundle install
         command: bundle update sprockets
```

That was failing with our `mail 2.8.0.rc1` dependency somehow. Removing it entirely caused all builds to fail, it really was required. But changing it to do an overall `bundle update`, not just sprockets, worked to get green build?

```
      - run:
         name: Update bundle to fix version incompatibilities between internal test app and initial bundle install
         command: bundle update
```

I'm honestly not sure what's going on here or what is considered samvera/engine_cart best practice. I hope I'm making things better not worse.  Any feedback is welcome, I'm not sure who might be available to provide some that might have more expertise here.

File: .circleci/config.yml [hunk 1]
```diff
@@ -36,11 +36,11 @@ jobs:
           ruby_version: << parameters.ruby_version >>
 
       - samvera/engine_cart_generate:
-          cache_key: v1-internal-test-app-{{ checksum "browse-everything.gemspec" }}-{{ checksum "Gemfile" }}-{{ checksum "spec/test_app_templates/lib/generators/test_app_generator.rb" }}-{{ checksum "lib/generators/browse_everything/install_generator.rb" }}-{{ checksum "lib/generators/browse_everything/config_generator.rb" }}--<< parameters.rails_version >>-<< parameters.ruby_version >>
+          cache_key: v1-internal-test-app-{{ checksum "browse-everything.gemspec" }}-{{ checksum "Gemfile" }}--{{ checksum "spec/test_app_templates/Gemfile.extra" }}-{{ checksum "spec/test_app_templates/lib/generators/test_app_generator.rb" }}-{{ checksum "lib/generators/browse_everything/install_generator.rb" }}-{{ checksum "lib/generators/browse_everything/config_generator.rb" }}--<< parameters.rails_version >>-<< parameters.ruby_version >>
```
Comment (jrgriffiniii) [inline .circleci/config.yml:39]: 👍

File: spec/test_app_templates/Gemfile.extra [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Only necessary until mail 2.8.0 is released, allow us to build with engine_cart
```
Comment (jrgriffiniii) [inline spec/test_app_templates/Gemfile.extra:1]: Excellent, thank you very much for finding this!

**Review by jrgriffiniii — APPROVED** [review]

**Pull request merged**